bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/AbstractRemoteAddonService.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    @Override
+    public void install(String id) {
+        Addon addon = getAddon(id, null);
+        if (addon != null) {
+            for (MarketplaceAddonHandler handler : addonHandlers) {
+                if (handler.supports(addon.getType(), addon.getContentType())) {
+                    if (!handler.isInstalled(addon.getId())) {
+                        try {
+                            handler.install(addon);
+                            installedAddonStorage.put(id, gson.toJson(addon));
+                            refreshSource();
+                            postInstalledEvent(addon.getId());
+                        } catch (MarketplaceHandlerException e) {
+                            postFailureEvent(addon.getId(), e.getMessage());
+                        }
+                    } else {
+                        postFailureEvent(addon.getId(), "Add-on is already installed.");
+                    }
+                    return;
+                }
+            }
+        }
+        postFailureEvent(id, "Add-on not known.");
+    }
+
+    @Override
+    public void uninstall(String id) {
+        Addon addon = getAddon(id, null);
+        if (addon != null) {
+            for (MarketplaceAddonHandler handler : addonHandlers) {
+                if (handler.supports(addon.getType(), addon.getContentType())) {
+                    if (handler.isInstalled(addon.getId())) {
+                        try {
+                            handler.uninstall(addon);
+                            installedAddonStorage.remove(id);
+                            refreshSource();
+                            postUninstalledEvent(addon.getId());
+                        } catch (MarketplaceHandlerException e) {
+                            postFailureEvent(addon.getId(), e.getMessage());
+                        }
+                    } else {
+                        installedAddonStorage.remove(id);
+                        postFailureEvent(addon.getId(), "Add-on is not installed.");
+                    }
+                    return;
+                }
+            }
+        }
+        postFailureEvent(id, "Add-on not known.");
+    }
+
+    @Override
+    public abstract @Nullable String getAddonId(URI addonURI);
+
+    /**
+     * check if remote services are enabled
+     *
+     * @return true if network access is allowed
+     */
+    protected boolean remoteEnabled() {
+        try {
+            Configuration configuration = configurationAdmin.getConfiguration("org.openhab.addons", null);
+            Dictionary<String, Object> properties = configuration.getProperties();
+            if (properties == null) {
+                // if we can't determine a set property, we use true (default is remote enabled)
+                return true;
+            }
+            Object remoteEnabled = properties.get("remote");
+            return (remoteEnabled == null || "true".equals(remoteEnabled.toString()));
+
```

Comment:
We can use the exposed `ConfigParser` here:

```suggestion
            return ConfigParser.valueAsOrElse(properties.get("remote"), Boolean.class, true);
```

Reply:
Done.